server/chat-plugins/blackjack.js

```diff
@@ -0,0 +1,643 @@
+			if (this.dealer.points < 17) {
+				this.giveCard('dealer');
+			} else if (this.dealer.points >= 17) {
+				let cards = '';
+				for (let card of this.dealer.cards) cards += `[${card}] `;
+				let turnLine = `<br /><strong>${this.dealer.name}</strong> stands with ${cards} (${this.dealer.points})`;
+				this.turnLog += turnLine;
+				this.display(turnLine);
+				this.end();
+			}
+			return;
+		}
+		if (this.playerTable[this.curUser].status !== 'playing') {
+			let number = 0;
+			while (this.playerTable[Object.keys(this.playerTable)[number]].status !== 'playing') number++;
+			this.curUser = Object.keys(this.playerTable)[number];
+		}
+		let output = `|uhtml${this.playerTable[this.curUser].selfUhtml}|user-blackjack-${this.blackjackNumber}|<div class="infobox" style="line-height: 20px;">`;
+		output += `It's your turn to move, ${this.playerTable[this.curUser].name}<br />`;
+		for (let card of this.playerTable[this.curUser].cards) {
+			output += this.generateCard(card);
+		}
+		output += `<br />Score: ${this.playerTable[this.curUser].points}${(this.playerTable[this.curUser].points === 21 ? ` (you have blackjack!)` : ``)}`;
+		output += `<br /><button class="button" name="send" value="/blackjack hit" title="Hit (get another card)">Hit</button> | <button class="button" name="send" value="/blackjack stand" title="Stand (just keep these cards)">Stand</button>`;
+
+		this.playerTable[this.curUser].sendRoom(`|notify|Blackjack (${this.room.title})|It's your turn to play!`);
+		this.playerTable[this.curUser].sendRoom(output);
+		this.playerTable[this.curUser].playScreen = output;
+
+		this.createTimer(this.playerTable[this.curUser]);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+		this.game = game;
+
+		this.cards = [];
+		this.points = 0;
+		this.slide = 0;
+		this.status = 'playing';
+
+		this.selfUhtml = '';
+		this.gameLog = '';
+		this.playScreen = '';
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60; // to get into seconds-format
+	}
+	resetTimerTicks() {
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60;
+	}
+}
+
+class BlackjackDealer {
+	constructor() {
+		this.cards = [];
+		this.points = 0;
+		this.name = 'The Dealer';
+	}
+}
+
+class BlackjackDeck {
+	constructor() {
+		this.deck = [
+			'A♥', 'A♦', 'A♣', 'A♠', '2♥', '2♦', '2♣', '2♠', '3♥', '3♦', '3♣',
+			'3♠', '4♥', '4♦', '4♣', '4♠', '5♥', '5♦', '5♣', '5♠', '6♥', '6♦', '6♣', '6♠',
+			'7♥', '7♦', '7♣', '7♠', '8♥', '8♦', '8♣', '8♠', '9♥', '9♦', '9♣', '9♠', '10♥',
+			'10♦', '10♣', '10♠', 'J♥', 'J♦', 'J♣', 'J♠', 'Q♥', 'Q♦', 'Q♣', 'Q♠', 'K♥', 'K♦',
+			'K♣', 'K♠',
+		];
+	}
+	shuffle() {
+		return Dex.shuffle(this.deck);
+	}
+}
+
+exports.commands = {
+	bj: 'blackjack',
+	blackjack: {
+		new: 'create',
+		create(target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
```

Comment:
`gamemanagement` is the "manage room settings related to games" permission, it's leader / gchat host by default, I believe `minigame` is what you're looking for.

Reply:
fixed